src/utils/classes/pattern.ts

```diff
@@ -0,0 +1,34 @@
+import { IPatternSettings } from "./models-interfaces";
+
+export class Pattern {
+	// metronome sound should not be here?
+	// private metronomeSound: MetronomeSounds = MetronomeSounds.MetronomeSoundOff;
+
+	private defaultPatternSettings: IPatternSettings = {
+		playNotes: 'quarterNotes',
+		playFillOn: { beat: '3', subBeat: '0' }, // can we make these objects into classes of their own or will it be overkill?
+		timeSignature: { beats: '4', division: '4' }, // can we make these objects into classes of their own or will it be overkill?
+	};;
+
+	private patternSettings: IPatternSettings; // user provided settings
+
+	constructor(patternSettings?: IPatternSettings) {
+		this.patternSettings = patternSettings ? { ...this.getDefaultPatternSettings(), ...patternSettings} : this.getDefaultPatternSettings(); 
```

Comment:
I'm not sure if I understand this line. Should it not read:
`this.patternSettings = patternSettings ? ...patternSettings : this.getDefaultPatternSettings();` ?
Why is it an object of two different pattern settings on the left side of the ternary?

Reply:
note: it's probably my understanding that is wrong, please let me know what's going on here :) I'm away from my home computer so don't have repo set up to properly debug etc.

Reply:
@dcallus so if you pass in a patternSettings object then it simply copies over the values from your object to class member. If you dont pass in one then it uses default settings to initialize the patternSettings.